lib/trino-record-decoder/src/main/java/io/trino/decoder/protobuf/ProtobufValueProvider.java

```diff
@@ -139,6 +142,15 @@ private static Slice getSlice(Object value, Type type, String columnName)
             return Slices.wrappedBuffer(((ByteString) value).toByteArray());
         }
 
+        if (type.getBaseName().equals(JSON) && value instanceof DynamicMessage) {
+            try {
+                return utf8Slice(JsonFormat.printer().print(((DynamicMessage) value)));
+            }
+            catch (InvalidProtocolBufferException e) {
+                throw new TrinoException(DECODER_CONVERSION_NOT_SUPPORTED, e);
+            }
+        }
+
```

Comment:
Json object convert is doing by very naive way to open discussion how to better implement it.
cc: @kokosing, @Praveen2112, @findepi